### ISSUE TITLE
sql like operator ?

### ISSUE THREAD

**Issue by ghost** [arangodb/arangodb#217]
In SQL you can filter result sets like this: `SELECT * FROM Persons WHERE City LIKE '%s'`
How can you do that (serverside) using ArangoDB?
I've not found any example or information regarding that kind of filtering. :)

**Comment by nikos90** [comment]
Any updates on this?

**Comment by CoDEmanX** [comment]
Full text search supports `whatever%` only (wildcard at the very end).

AQL's `LIKE()` allows SQL-style queries with any amount of `%` and `_` anywhere in the search phrase, but what is does internally is to create a regular expression and match every document against it. Hence, none of the possibly existing indexes is used. Performance is not too bad however - I created 100k docs with random tokens of variable length (up to 1k) as sample text:

    FOR i IN 1..100000
        INSERT {_key: TO_STRING(i), text: RANDOM_TOKEN(RAND() * 1000 + 1)} INTO test

Then queried for some pattern:

    FOR doc IN test
        FILTER LIKE(doc.text, "%aran_o%", true)
        RETURN doc.text

Found 3 matches in ~3 seconds. If you query for something like `%a%b%c%d%e%f%g%` however, it will take forever. That's a typical regex issue and [Boyer-Moore][2]  - which is used in MySQL AFAIK - would probably outperform it by orders of magnitude. It's a rare corner case on the other hand.

Specs: Run on MacBook Air, Intel i5-4260U (up to 2.52Ghz in Turbo mode), 4GB RAM, SSD, Windows 8.1 64bit, ArangoDB 2.7.0-devel ([custom build][1]).

+1 for adding index-supported full text `*suffix` and `*contains*` searches. Maybe even think about to implement the aforementioned fuzzy string matching algorithm?

[1]: https://github.com/CoDEmanX/ArangoDB/releases/tag/d5c7d596
[2]: https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore_string_search_algorithm
